### ISSUE TITLE
add topics/cancel end point

### ISSUE THREAD

**Issue by hroberts** [berkmancenter/mediacloud#552]
add topics/cancel end point to end any currently running spider or snapshot jobs for a topic.

**Comment by cindyloo** [comment]
I think this is essentially done - new versions replace old versions, effectively cancelling them. We don't have an api endpoint however

**Comment by hroberts** [comment]
This is not really true.  The idea of the cancel end point is that it would
actually stop whatever topic spidering or snapshotting jobs are happening
on the back end.  Even if you create a new version, the system will
continue trying to run the old version.

**Comment by cindyloo** [comment]
ok, I think we were addressing two concerns. 1) as far as the researchers are concerned, they wanted a way to stop/cancel/start over with a topic. Versioning does take care of that. 2) a way to stop the spidering/snapshotting jobs. What I didn't know is that with a new version, we are still running back end jobs from previous versions... Maybe that is what should be addressed, and surfacing that to the admin user

**Comment by hroberts** [comment]
I still don't think this is true.  spidering and snapshotting jobs both
have locks that will prevent more than one job running at a time for a
given topic.  so if the front creates a new version and starts the
spider/snapshot job, then creates a second new version and starts a
spider/snapshot job for the second one, the second job will fail if the
first one is still running.

-hal

**Comment by cindyloo** [comment]
gotcha. 
so if a version is `completed`, no jobs are running on the back end correct? Creating a new version spiders/creates a new snapshot with modified info. Older versions should be described as ___ (not cancelled)?

if a version isn't completed/still running/spidering, in the UI, we allow admins to create new version but not regular users. If a new version is created, the newer one will fail in all cases (for admins and users).

so then surfacing a cancellation is still needed for admins at least. none of this is particularly obvious to the front-end users.

**Comment by hroberts** [comment]
yes, once a version is completed, nothing is running on the back end.
 older versions should just be described as whatever their state is
(queued, running, completed, etc).

this should all be a pretty rare use case, and the bad behavior when the
lock gets hit is that the new version just fails with an error that says
something like 'failed to get lock for snapshot for topic 3290', so I don't
think we should worry about the edge case unless folks start hitting it.  I
just don't want you to build the ui to affirmatively give the user the
notion that she can cancel a running job.


-hal
